lib/presentation/widgets/task/task_list_item.dart

```diff
@@ -0,0 +1,104 @@
+import 'package:flutter/material.dart';
+import 'package:vikunja_app/domain/entities/task.dart';
+import 'package:vikunja_app/presentation/widgets/due_date_card.dart';
+import 'package:vikunja_app/presentation/widgets/project/kanban/priority_batch.dart';
+
+class TaskListItem extends StatefulWidget {
+  final Task task;
+  final Function onTap;
+  final Function onEdit;
+  final Function(bool value) onCheckedChanged;
+
+  const TaskListItem({
+    super.key,
+    required this.task,
+    required this.onTap,
+    required this.onEdit,
+    required this.onCheckedChanged,
+  });
+
+  @override
+  TaskListItemState createState() => TaskListItemState();
+}
+
+class TaskListItemState extends State<TaskListItem> {
+  TaskListItemState();
+
+  @override
+  Widget build(BuildContext context) {
+    var isThreeLine =
+        widget.task.hasDueDate ||
+        widget.task.priority != null && widget.task.priority != 0;
+
+    return Stack(
+      fit: StackFit.loose,
+      children: [
+        ListTile(
+          onTap: () {
+            widget.onTap();
+          },
+          title: Text(
+            widget.task.title,
+            maxLines: 1,
+            overflow: TextOverflow.ellipsis,
+          ),
+          subtitle: _buildTaskSubtitle(widget.task, context),
+          leading: Checkbox(
+            value: widget.task.done,
+            onChanged: (bool? newValue) {
+              if (newValue != null) {
+                widget.onCheckedChanged(newValue);
+              }
+            },
+          ),
+          trailing: IconButton(
+            icon: Icon(Icons.edit),
+            onPressed: () => widget.onEdit(),
+          ),
+        ),
+        Container(
+          width: 4.0,
+          height: isThreeLine ? 88.0 : 72.0,
+          color: widget.task.color,
+        ),
+      ],
+    );
+  }
+
+  Widget? _buildTaskSubtitle(Task task, BuildContext context) {
+    List<Widget> texts = [];
+
+    if (task.hasDueDate) {
+      texts.add(
+        Padding(
+          padding: const EdgeInsets.only(right: 8.0),
+          child: DueDateCard(task.dueDate!),
+        ),
+      );
+    }
+    if (task.priority != null && task.priority != 0) {
+      texts.add(PriorityBatch(task.priority!));
+    }
+
+    if (texts.isEmpty) {
+      return Text(task.project?.title ?? "-");
```

Comment:
Maybe we don't need to show a `-` when no project is available?

Reply:
It seems like now the project is only shown some times:

![Screenshot_20251107-125910.png](https://github.com/user-attachments/assets/81cfef24-f118-4899-9cef-6d9a7b433e20)

What are cases where we don't have the project? Is this a case where the API does not return it or do we need to load more data somehow?